lms/djangoapps/bulk_email/tasks.py

```diff
@@ -97,25 +97,31 @@
 )
 
 
-def _get_course_email_context(course):
+def _get_course_email_context(course, is_secure):
     """
     Returns context arguments to apply to all emails, independent of recipient.
+
+    Inputs are:
+    * `course`: Course related to the current email
+    * `is_secure`: Set to True to return https URLs, False to use http.
     """
     course_id = course.id.to_deprecated_string()
     course_title = course.display_name
     course_end_date = get_default_time_display(course.end)
-    course_url = 'https://{}{}'.format(
-        settings.SITE_NAME,
+    scheme = u'https' if is_secure else u'http'
```

Comment:
NB: I tried just using `settings.HTTPS` to determine http or https (as [used in a few places in edx-platform](https://github.com/edx/edx-platform/search?utf8=%E2%9C%93&q=%22settings.HTTPS%22&type=Code)), but that didn't work.  It appears that `settings.HTTPS` is always `True`.